main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/MonoDevelopWorkspace.cs

```
@@ -1366,7 +1366,9 @@ void OnFileRemoved (object sender, MonoDevelop.Projects.ProjectFileEventArgs arg
 					var id = data.GetDocumentId (fargs.ProjectFile.FilePath);
 					if (id != null) {
 						ClearDocumentData (id);
-						OnDocumentRemoved (id);
+						try {
+							OnDocumentRemoved (id);
+						} catch (Exception) {}

```

Please don't do use empty catch blocks. You should log the exception or, if the exception is expected, do a specialized catch.